src/config.rs

```diff
@@ -37,188 +39,55 @@ impl Default for Config {
-        self.blink_fanout = blink_fanout;
-    }
-
-    /// Set page_consolidation_threshold
-    pub fn set_page_consolidation_threshold(&mut self, threshold: usize) {
-        self.page_consolidation_threshold = threshold;
-    }
-
-    /// Set path
-    pub fn set_path(&mut self, path: Option<String>) {
-        self.path = path;
-    }
-
-    /// Set the number of bits used for addressing cache shards.
-    pub fn set_cache_bits(&mut self, cache_bits: usize) {
-        self.cache_bits = cache_bits;
-    }
-
-    /// Set the cache capacity in bytes.
-    pub fn set_cache_capacity(&mut self, cache_capacity: usize) {
-        self.cache_capacity = cache_capacity;
-    }
-
-    /// Set whether the system uses the OS pagecache for
-    /// compressed pages.
-    pub fn set_use_os_cache(&mut self, use_os_cache: bool) {
-        self.use_os_cache = use_os_cache;
-    }
-
-    /// Set whether the system compresses data written to
-    /// secondary storage with the zstd algorithm.
-    pub fn set_use_compression(&mut self, use_compression: bool) {
-        self.use_compression = use_compression;
-    }
-
-    /// Set the number of milliseconds between flushes.
-    pub fn set_flush_every_ms(&mut self, flush_every_ms: Option<u64>) {
-        self.flush_every_ms = flush_every_ms;
-    }
-
-    /// Builder, set number of io buffers
-    pub fn io_bufs(&self, io_bufs: usize) -> Config {
-        let mut ret = self.clone();
-        ret.io_bufs = io_bufs;
-        ret
-    }
-
-    /// Builder, set the max io buffer size
-    pub fn io_buf_size(&self, io_buf_size: usize) -> Config {
-        let mut ret = self.clone();
-        ret.io_buf_size = io_buf_size;
-        ret
-    }
-
-    /// Builder, set the b-link tree node fanout
-    pub fn blink_fanout(&self, blink_fanout: usize) -> Config {
-        let mut ret = self.clone();
-        ret.blink_fanout = blink_fanout;
-        ret
-    }
-
-    /// Builder, set the pagecache consolidation threshold
-    pub fn page_consolidation_threshold(&self, threshold: usize) -> Config {
-        let mut ret = self.clone();
-        ret.page_consolidation_threshold = threshold;
-        ret
-    }
-
-    /// Builder, set the filesystem path
-    pub fn path(&self, path: Option<String>) -> Config {
-        let mut ret = self.clone();
-        ret.path = path;
-        ret
-    }
-
-    /// Builder, set the number of bits used for addressing cache shards.
-    pub fn cache_bits(&self, cache_bits: usize) -> Config {
-        let mut ret = self.clone();
-        ret.cache_bits = cache_bits;
-        ret
-    }
-
-    /// Builder, set the cache capacity in bytes.
-    pub fn cache_capacity(&self, cache_capacity: usize) -> Config {
-        let mut ret = self.clone();
-        ret.cache_capacity = cache_capacity;
-        ret
-    }
+macro_rules! builder {
+    ($(($name:ident, $get:ident, $set:ident, $t:ty, $desc:expr)),*) => {
+        $(
+            #[doc="Get "]
+            #[doc=$desc]
```

Comment:
the macro system is sort of black magic, and pretty annoying, but in cases like this it can reduce tedium significantly.